chia/wallet/trade_manager.py

```diff
@@ -517,15 +524,26 @@ async def _create_offer_for_ids(
                     else:
                         asset_id = id
                         wallet = await self.wallet_state_manager.get_wallet_for_asset_id(asset_id.hex())
+                    assert wallet is not None
                     if not callable(getattr(wallet, "get_coins_to_offer", None)):  # ATTENTION: new wallets
                         raise ValueError(f"Cannot offer coins from wallet id {wallet.id()}")
                     # For the XCH wallet also include the fee amount to the coins we use to pay this offer
                     amount_to_select = abs(amount)
                     if wallet.type() == WalletType.STANDARD_WALLET:
                         amount_to_select += fee
-                    coins_to_offer[id] = await wallet.get_coins_to_offer(
-                        asset_id, uint64(amount_to_select), tx_config.coin_selection_config
-                    )
+                    assert isinstance(wallet, (CATWallet, DataLayerWallet, NFTWallet, Wallet))
+                    if isinstance(wallet, DataLayerWallet):
+                        assert asset_id is not None
+                        coins_to_offer[id] = await wallet.get_coins_to_offer(launcher_id=asset_id)
+                    elif isinstance(wallet, NFTWallet):
+                        assert asset_id is not None
+                        coins_to_offer[id] = await wallet.get_coins_to_offer(nft_id=asset_id)
```

Comment:
what's the reason for this change? do the wallets not have the same function signature for `get_coins_to_offer()`? It looks like `NFTWallet` and `DataLayerWallet` both take 1 argument, whereas the other take 3.

It seems like that intention was for these to use the same signature, which I would expect to be a more appropriate fix for this

Reply:
Yes, `get_coins_to_offer`'s signature is not the same across wallets, and `asset_id` also differs in terms of optionality. I also used this as a chance to make the param name clearer (`launcher_id` for one wallet and `nft_id` for another).

Reply:
it seems to go against the original design though; to have these functions implement a common interface.
If we truly want to treat these as different functions, they should probably have different names as well. But I still think the first attempted solution should be to implement the (presumed) original intent. i.e. make them have the same signature. Maybe even enforce the interface with the protocol.

Reply:
I share Quex's sentiment that issues like this are representative of a larger problem that is beyond the type annotations scope of this PR.